### PR TITLE
feat: webhook destination: introduce batching

### PR DESCRIPTION
**Pull request by aanshi07**
## What are the changes introduced in this PR?

This PR introduces changes to add batching functionality for events in the webhook destination .

## What is the related Linear task?

Resolves INT-2119

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3404/test-report.html

File: test/integrations/destinations/webhook/processor/data.ts [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+  destinationWithoutHeaders,
+} from '../commonConfig';
+import { generalProperties, userProperties } from '../router/basicProperties';
+
 export const data = [
   {
     name: 'webhook',
-    description: 'Test 0',
+    id: 'Test 0 - processor',
+    description: 'Basic track calls',
```
Comment (anantjain45823) [inline test/integrations/destinations/webhook/processor/data.ts:13]:
```suggestion
    description: 'Basic track call',
```

File: test/integrations/destinations/webhook/processor/data.ts [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
+    scenario: 'Framework',
+    successCriteria: 'All events should be transformed successfully and status code should be 200',
```
Comment (anantjain45823) [inline test/integrations/destinations/webhook/processor/data.ts:15]:
```suggestion
    successCriteria: 'Event should be transformed successfully and status code should be 200',
```

File: test/integrations/destinations/webhook/router/basicProperties.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+export const generalProperties = {
```
Comment (anantjain45823) [inline test/integrations/destinations/webhook/router/basicProperties.ts:1]: move this file a level above

File: test/integrations/destinations/webhook/processor/data.ts [hunk 4]
```diff
@@ -252,7 +90,10 @@ export const data = [
   },
   {
     name: 'webhook',
-    description: 'Test 1',
+    id: 'Test 1 - processor',
+    description: 'Basic track calls',
+    scenario: 'Framework',
+    successCriteria: 'All events should be transformed successfully and status code should be 200',
```
Comment (anantjain45823) [inline test/integrations/destinations/webhook/processor/data.ts:96]: Same as above

File: test/integrations/destinations/webhook/processor/data.ts [hunk 5]
```diff
@@ -252,7 +90,10 @@ export const data = [
   },
   {
     name: 'webhook',
-    description: 'Test 1',
+    id: 'Test 1 - processor',
+    description: 'Basic track calls',
```
Comment (anantjain45823) [inline test/integrations/destinations/webhook/processor/data.ts:94]: Track -> No headers in config object

**Comment by aanshi07** [comment]
https://github.com/coderabbitai.

File: test/integrations/destinations/webhook/commonConfig.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  os: { name: 'android', version: '8.1.0' },
+  traits: {
+    address: { city: 'Dhaka', country: 'Bangladesh' },
+    anonymousId: 'c82cbdff-e5be-4009-ac78-cdeea09ab4b1',
+  },
+};
+
+export const destination = {
+  Config: {
+    webhookUrl: 'http://6b0e6a60.ngrok.io',
```
Comment (yashasvibajpai) [inline test/integrations/destinations/webhook/commonConfig.ts:18]: Let's create an abstraction of a destination and utilise in others like destinationWithoutHeaders, destinationWithWrongUrl

File: test/integrations/destinations/webhook/basicProperties.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
+export const generalProperties = {
+  additional_bet_index: 0,
```
Comment (yashasvibajpai) [inline test/integrations/destinations/webhook/basicProperties.ts:2]: Do we need so many generalProperties here?

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Pull request closed without merging**